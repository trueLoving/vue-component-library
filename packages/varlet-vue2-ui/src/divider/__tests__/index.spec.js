import VarDivider from '../Divider'
import Vue from 'vue'
import example from '../example'
import { mount } from '@vue/test-utils'

test('test divider plugin', () => {
  Vue.use(VarDivider)
  expect(Vue.component(VarDivider.name)).toBeTruthy()
})

test('test divider example', () => {
  const wrapper = mount(example)
  expect(wrapper.html()).toMatchSnapshot()
  wrapper.destroy()
})

test('test divider inset and vertical prop', async () => {
  const wrapper = mount(VarDivider, {
    propsData: {
      inset: true,
    },
  })

  expect(wrapper.html()).toMatchSnapshot()

  await wrapper.setProps({
    inset: 36,
  })
  expect(wrapper.html()).toMatchSnapshot()

  await wrapper.setProps({
    inset: '-36px',
  })
  expect(wrapper.html()).toMatchSnapshot()

  await wrapper.setProps({
    inset: '-36px',
    vertical: true,
  })
  expect(wrapper.html()).toMatchSnapshot()
  wrapper.destroy()
})

test('test divider dashed and margin prop', async () => {
  const wrapper = mount(VarDivider)

  expect(wrapper.html()).toMatchSnapshot()

  await wrapper.setProps({
    dashed: true,
  })
  expect(wrapper.html()).toMatchSnapshot()

  await wrapper.setProps({
    dashed: false,
    margin: '36px 0',
  })
  expect(wrapper.html()).toMatchSnapshot()
  wrapper.destroy()
})

test('test divider description prop', () => {
  const desc = 'description'
  const wrapper = mount(VarDivider, {
    propsData: {
      description: desc,
    },
  })
  expect(wrapper.text()).toBe(desc)
  wrapper.destroy()
})

test('test default slot of divider', () => {
  const wrapper = mount({
    components: {
      [VarDivider.name]: VarDivider,
    },
    template: `
      <var-divider>
        <span>description</span>
      </var-divider>
    `,
  })
  expect(wrapper.html()).toMatchSnapshot()
  wrapper.destroy()
})
