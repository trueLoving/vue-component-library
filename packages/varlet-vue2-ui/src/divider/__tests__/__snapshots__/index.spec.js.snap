// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test default slot of divider 1`] = `"<div class=\\"var-divider var--box\\"><span>description</span></div>"`;

exports[`test divider dashed and margin prop 1`] = `
"<div class=\\"var-divider var--box\\">
  <!---->
</div>"
`;

exports[`test divider dashed and margin prop 2`] = `
"<div class=\\"var-divider var--box var-divider--dashed\\">
  <!---->
</div>"
`;

exports[`test divider dashed and margin prop 3`] = `
"<div class=\\"var-divider var--box\\" style=\\"margin: 36px 0px;\\">
  <!---->
</div>"
`;

exports[`test divider example 1`] = `
"<div class=\\"var-divider-example\\">
  <div>
    <div class=\\"app-type\\">基本使用</div>
    <div class=\\"var-divider var--box\\">
      <!---->
    </div>
  </div>
  <div>
    <div class=\\"app-type\\">虚线</div>
    <div class=\\"var-divider var--box var-divider--dashed\\">
      <!---->
    </div>
  </div>
  <div>
    <div class=\\"app-type\\">缩进</div>
    <div class=\\"var-divider var--box var-divider--inset\\">
      <!---->
    </div>
    <div class=\\"var-divider var--box var-divider--inset\\" style=\\"margin: 36px 0px; width: calc(100% - 36px); left: 36px;\\">
      <!---->
    </div>
    <div class=\\"var-divider var--box var-divider--inset\\" style=\\"width: calc(100% - 36px); left: 0px;\\">
      <!---->
    </div>
  </div>
  <div>
    <div class=\\"app-type\\">垂直分割线</div>
    <div class=\\"vertical-divider-wrapper\\"><span>文字</span>
      <div class=\\"var-divider var--box var-divider--vertical\\">
        <!---->
      </div> <span>文字</span>
      <div class=\\"var-divider var--box var-divider--vertical\\">
        <!---->
      </div> <span>文字</span></div>
  </div>
  <div>
    <div class=\\"app-type\\">带有文字描述的分割线</div>
    <div class=\\"var-divider var--box\\"><span class=\\"var-divider__text\\">文字描述</span></div>
  </div>
  <div>
    <div class=\\"app-type\\">自定义</div>
    <div class=\\"var-divider var--box\\"><i class=\\"var-icon var-icon--set var-icon-heart-outline\\" style=\\"color: rgb(41, 121, 255); transition: transform 0ms; margin: 0px 16px;\\"></i></div>
  </div>
</div>"
`;

exports[`test divider inset and vertical prop 1`] = `
"<div class=\\"var-divider var--box var-divider--inset\\">
  <!---->
</div>"
`;

exports[`test divider inset and vertical prop 2`] = `
"<div class=\\"var-divider var--box var-divider--inset\\" style=\\"width: calc(100% - 36px); left: 36px;\\">
  <!---->
</div>"
`;

exports[`test divider inset and vertical prop 3`] = `
"<div class=\\"var-divider var--box var-divider--inset\\" style=\\"width: calc(100% - 36px); left: 0px;\\">
  <!---->
</div>"
`;

exports[`test divider inset and vertical prop 4`] = `
"<div class=\\"var-divider var--box var-divider--vertical var-divider--inset\\" style=\\"height: calc(80% - 36px);\\">
  <!---->
</div>"
`;
